parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -524,6 +525,11 @@ public Optional<LogicalType> visit(LogicalTypeAnnotation.UnknownLogicalTypeAnnot
     public Optional<LogicalType> visit(LogicalTypeAnnotation.IntervalLogicalTypeAnnotation intervalLogicalType) {
       return of(LogicalType.UNKNOWN(new NullType()));
     }
+
+    @Override
+    public Optional<LogicalType> visit(LogicalTypeAnnotation.VariantLogicalTypeAnnotation variantLogicalType) {
+      return of(LogicalTypes.VARIANT());
```

Comment:
The other cases create new instances of the logical type, but I agree with using the `LogicalTypes` constant. This just doesn't need to be a method call. It would also be nice to follow up with a PR that converts all of these to the correct constant rather than creating new instances.

Reply:
Will do.

Reply:
It has been addressed in https://github.com/apache/parquet-java/pull/3186.